lib/like-selector.js

```diff
@@ -1,8 +1,14 @@
 export function isLikeSelector(selector) {
-	return selector !== null
-		&& typeof selector === 'object'
-		&& Reflect.getPrototypeOf(selector) === Object.prototype
-		&& Reflect.ownKeys(selector).length > 0;
+	const isArrayOrObject = selector !== null && typeof selector === 'object';
+
+	if (!isArrayOrObject) {
+		return false;
+	}
+
+	const isArray = Array.isArray(selector);
+	const minimumKeysRequired = isArray ? 1 : 0;
+
+	return Reflect.ownKeys(selector).length > minimumKeysRequired;
```

Comment:
I think we need to keep the check for `Object.prototype`, otherwise you could pass other non-array values like functions or maps.

Something like this may be more elegant (but probably doesn't satisfy the linter):

```suggestion
	if (selector === null || typeof selector !== 'object') return false;
	
	const keyCount = Reflect.ownKeys(selector).length;
	return Array.isArray(selector) && keyCount > 1 || Reflect.getPrototypeOf(selector) === Object.prototype && keyCount > 0;
```

Reply:
That's a much cleaner solution, there wasn't much to change for the linter.